src/Plugin/resource/Field/ResourceFieldEntity.php

```diff
@@ -890,6 +890,10 @@ public static function fieldClassName(array $field_definition) {
       case 'text':
       case 'text_long':
       case 'text_with_summary':
+        // If the field is treated as a resource, then default to the reference.
+        if (!empty($field_definition['resource'])) {
```

Comment:
:weary: I had to strip this out from an ongoing PR. This should go to the File field.

Good :eyes: